controllers/data.controller.js

```diff
@@ -34,3 +36,20 @@ exports.getArticles = (req, res, next) => {
       next(err);
     });
 };
+
+exports.getComments = (req, res, next) => {
+  const id = req.params.article_id;
+
+  const commentPromises = [fetchComments(id)];
+  if (id) {
+    commentPromises.push(checkArticleExists(id));
+  }
```

Comment:
Just a little optional refac -
Is there ever not an id?
Because theres always an id (its part of the route) you could lose the if block and just put the checkArticleExists in the array to begin with